src/interface/src/app/map/map-manager.ts

```diff
@@ -673,6 +675,56 @@ export class MapManager {
     }
   }
 
+  addLegend(colormap: any, map: Map){
```

Comment:
nit: spacing (e.g. "Map){" vs "Map) {") is inconsistent with the other parts of this file (and other files).